### DISCUSSION TITLE
Mavlink not connecting

### DISCUSSION THREAD

**Discussion by ievbayiro32** [DroneEngage/droneengage_mavlink#4] [General]
Hello,

I'm experiencing a minor issue with my Pi and the pixhawk for my senior design project. Our mavlink isn't connecting to the web client, even though we enabled UART and disabled bluetooth

This is the error in the ardupilot.so file in the mavlink folder

=================== STARTING PLUGIN ===================

Drone-Engage FCB Module version 3.7.1
Thu Apr 18 18:00:42 2024
1713459642 seconds since the Epoch
Read config file: de_mavlink.config.module.json ....
Read internal config file: de_mavlink.local .... succeeded
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.302] type must be number, but is null
Aborted
pi@raspberrypi:~/drone_engage/de_mavlink $

Here's a [screenshot](
![Capture (1)](https://github.com/DroneEngage/droneengage_mavlink/assets/165714210/a30955ad-1747-491a-b4c4-8a36427b0048)
![Capture2](https://github.com/DroneEngage/droneengage_mavlink/assets/165714210/4aeeb326-4a5a-4c75-abda-f07095e4e1ac)
![Capture3](https://github.com/DroneEngage/droneengage_mavlink/assets/165714210/a5e98da5-202d-4f08-88b3-e49d1f5b83ae)) of the data received from cu command attached

**Comment by HefnySco** [comment]
Please paste config file text here or attached it  so that I can try it and detect the issue.

**Reply by ievbayiro32** [reply]
Here is is pasted below
de_mavlink.config.module.json:



{
  "module_id": "FCB_CTRL",
  "module_key": "E289FEE7-F4AD-44EF-A257-C9A361316BE7",
  "s2s_udp_target_ip": "127.0.0.1",
  "s2s_udp_target_port": "60000",
  "s2s_udp_listening_ip": "127.0.0.1",
  "s2s_udp_listening_port": "60003",
  
  "fcbConnectionURI":
  {
   "type": "serial",
   "port": "/dev/ttyAMA0",
   "baudrate": 115200
  },

  "logger_enabled": false,
  "logger_debug": false,
  "default_optimization_level": 2,
  "udp_proxy_enabled": true,
  "read_only_mode": false,
  "ignore_loading_parameters": false,
  "message_timeouts": {
    "1": [
      0,
      250,
      500,
      1000
    ],
    "2": [
      0,
      250,
      500,
      1000
    ],
    "24": [
      0,
      800,
      1000,
      2000
    ],
    "27": [
      0,
      500,
      1000,
      2000
    ],
    "28": [
      0,
      500,
      1000,
      2000
    ],
    "29": [
      0,
      500,
      2000,
      4000
    ],
    "30": [
      0,
      250,
      1000,
      2000
    ],
    "32": [
      0,
      250,
      1000,
      2000
    ],
    "33": [
      0,
      250,
      1000,
      2000
    ],
    "34": [
      0,
      500,
      1000,
      2000
    ],
    "35": [
      0,
      500,
      1000,
      2000
    ],
    "36": [
      0,
      1000,
      2000,
      2000
    ],
    "42": [
      0,
      1000,
      2000,
      4000
    ],
    "62": [
      0,
      250,
      500,
      1000
    ],
    "65": [
      0,
      500,
      1000,
      2000
    ],
    "74": [
      0,
      500,
      1000,
      2000
    ],
    "87": [
      0,
      500,
      1000,
      1000
    ],
    "116": [
      0,
      1000,
      2000,
      4000
    ],
    "124": [
      0,
      800,
      1000,
      2000
    ],
    "125": [
      0,
      1000,
      2000,
      4000
    ],
    "129": [
      0,
      1000,
      2000,
      4000
    ],
    "136": [
      0,
      500,
      1000,
      2000
    ],
    "137": [
      0,
      500,
      2000,
      4000
    ],
    "143": [
      0,
      500,
      2000,
      4000
    ],
    "147": [
      0,
      500,
      2000,
      4000
    ],
    "152": [
      0,
      4000,
      8000,
      12000
    ],
    "163": [
      0,
      250,
      500,
      1000
    ],
    "164": [
      0,
      2000,
      4000,
      12000
    ],
    "165": [
      0,
      250,
      1000,
      2000
    ],
    "168": [
      0,
      1000,
      2000,
      4000
    ],
    "178": [
      0,
      250,
      500,
      1000
    ],
    "182": [
      0,
      250,
      500,
      1000
    ],
    "193": [
      0,
      250,
      500,
      1000
    ],
    "241": [
      0,
      250,
      1000,
      2000
    ],
    "234": [
      0,
      250,
      1000,
      2000
    ],
    "235": [
      0,
      250,
      1000,
      2000
    ],
    "11030": [
      0,
      1000,
      2000,
      5000
    ],
    "11031": [
      0,
      1000,
      2000,
      5000
    ],
    "11032": [
      0,
      1000,
      2000,
      5000
    ]
  },
  "rc_block_channel": 0,
  "rc_channels": {
    "rc_smart_channels": {
      "active": true,
      "rc_channel_enabled": [
        1,
        1,
        1,
        1
      ],
      "rc_channel_limits_max": [
        2000,
        2000,
        1750,
        2000
      ],
      "rc_channel_limits_min": [
        1000,
        1000,
        1300,
        1000
      ]
    },
    "rc_channel_enabled": [
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1
    ],
    "rc_channel_reverse": [
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1,
      1
    ],
    "rc_channel_limits_max": [
      1850,
      2000,
      1750,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000,
      2000
    ],
    "rc_channel_limits_min": [
      1150,
      1000,
      1300,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000,
      1000
    ]
  },
  "event_fire_channel": null,
  "event_wait_channel": null
}

**Comment by HefnySco** [comment]
seems that installation script has an issue.
Please remove 
"event_fire_channel": null,
"event_wait_channel": null
or set them to 0
"event_fire_channel": 0,
"event_wait_channel": 0

**Comment by HefnySco** [comment]
These parameters are related to mission synchronization between drones.
[https://youtu.be/YwEw_YSFVEo?si=DOLCpLNcUczxpHma](https://youtu.be/YwEw_YSFVEo?si=DOLCpLNcUczxpHma)

**Comment by HefnySco** [comment]
ALso please  rename **fcbConnectionURI** tp 
"**fcb_connection_uri**"
